lib/pf/cmd/pf/ipmachistory.pm

```
@@ -1,45 +1,83 @@
+Get the IP/MAC mapping history for a specific IP or MAC with optional date range (in MySQL format)
+
+Syntax: pfcmd ipmachistory <ip|mac> [start_time=<date> end_time=<date>] [limit=<limit>]
+
+Where:
+
+ - <ip|mac> either the IP or the MAC address for which we want the IP/MAC mapping history (REQUIRED)
+
+ - [start_time=<date> end_time=<date>] start_time and end_time (in MySQL format) for optional date range (OPTIONAL)
 
-get the MAC/IP mapping for a specified IP or MAC with optional date (in mysql format)
+ - [limit=<limit>] limit the number of returned results. If not specified, all results will be returned (OPTIONAL)
+
+Examples:
 
-examples:
   pfcmd ipmachistory 192.168.1.100
-  pfcmd ipmachistory 192.168.1.100 start_time=2006-10-12 15:00:00,end_time=2006-10-18 12:00:00
+
+  pfcmd ipmachistory de:ad:be:ef:00:42
+
+  pfcmd ipmachistory 192.168.1.100 start_time="2006-10-12 15:00:00" end_time="2006-10-18 12:00:00"
+
+  pfcmd ipmachistory 192.168.1.100 limit=42
 
 =head1 DESCRIPTION
 
-pf::cmd::pf::ipmachistory
+Get the IP/MAC mapping history for a specific IP or MAC with optional date range (in MySQL format)
 
 =cut
 
 use strict;
 use warnings;
-use base qw(pf::cmd::display);
 
 use Date::Parse;
 
+use base qw(pf::cmd::display);
+
 sub parseArgs {
-    my ($self) = @_;
-    my ($key,$start_time,$end_time) = $self->args;
-    if($key) {
+    my ( $self ) = @_;
+
+    my ( $key, @params ) = $self->args;
+
+    my %params = (
+        'start_time'    => undef,
+        'end_time'      => undef,
+        'limit'         => undef,
+    );
+    foreach my $param ( @params ) {
+        my @data = split('=', $param);
+        if ( exists($params{$data[0]}) ) {
+            if ( length($data[1]) >= 1 ) {
+                $params{$data[0]} = $data[1];
+            }
+            else {
+                print "Invalid parameter value '$data[1]' for parameter '$data[0]'";
+            }
+        }
+        else {
+            print "Unknown parameter '$data[0]'";
```

You should return an error here
And print to STDERR